### PR TITLE
Hotspot listing

### PR DESCRIPTION
**Pull request by marcheing**

### CODE REVIEW

File: app/assets/javascripts/module/tree.js.coffee [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class Module.Tree
   @load: (loading_html, module_id) ->
+    # FIXME: The messages we send on loading_html are the same already
```
Comment (rafamanzo) [inline app/assets/javascripts/module/tree.js.coffee]: Not sure if this is a FIXME. I believe that this load message setter is there for the case where the user ask to reprocess or ask for a processing with specific date.

What do you think? Can we remove it?

File: features/tree_metric_configuration/edition.feature [hunk 2]
```diff
@@ -9,9 +9,9 @@ Feature: Metric Configuration edition
     And I am signed in
     And I have a sample configuration
     And I have a sample reading group
-    And I have a sample metric configuration within the given mezuro configuration
+    And I have a sample tree metric configuration within the given mezuro configuration
     When I am at the Sample Configuration page
-    Then I should not see Edit within table
+    Then I should not see Edit within table#tree_metric_configurations
```
Comment (rafamanzo) [inline features/tree_metric_configuration/edition.feature]: Does this step do work? I think cucumber handle `#` as comment.

**Comment by rafamanzo** [comment]
Nice job team :trophy: 

Mode just two minor questions/suggestions.

Waiting for everyone else review :smile:

File: config/locales/views/repository/pt.yml [hunk 3]
```diff
@@ -26,6 +26,7 @@ pt:
     errors:
       repository:
         no_metric_results: "O processamento do Repósitório retornou um erro. Não há Resultados de Métrica."
+        no_hotspot_metric_results: "O processamento do Repósitório retornou um erro. Não há Resultados de Métricas de Hotspot."
         no_modeule_tree: "Repository process returned with error. Não há uma Árvore de Módulos."
```
Comment (dread-uo) [inline config/locales/views/repository/pt.yml]: This should be `no_module_tree`, right?

File: config/locales/views/repository/en.yml [hunk 4]
```diff
@@ -25,7 +25,8 @@ en:
         branch: "The branch to be analyzed."
     errors:
       repository:
-        no_metric_results: "Repository process returned with error. There are no metric results."
+        no_metric_results: "Repository process returned with error. There are no tree metric results."
+        no_hotspot_metric_results: "Repository process returned with error. There are no hotspot metric results."
         no_modeule_tree: "Repository process returned with error. There is no module tree."
```
Comment (dread-uo) [inline config/locales/views/repository/en.yml]: This should be `no_module_tree`, right?

File: config/locales/views/repository/pt.yml (hunk 3 above)
Comment (dread-uo) [inline config/locales/views/repository/pt.yml]: This should be `no_module_tree`, as well?

File: features/tree_metric_configuration/edition.feature (hunk 2 above)
Comment (marcheing) [inline features/tree_metric_configuration/edition.feature]: It really does work! Actually, without this selector, the step would find 2 tables and would be unable to continue

File: app/assets/javascripts/module/tree.js.coffee (hunk 1 above)
Comment (marcheing) [inline app/assets/javascripts/module/tree.js.coffee]: I'm not sure. Even if the user ask for a reprocess, the page still loads the correct div anyway.

**Comment by diegoamc** [comment]
@dread-uo have you reviewed this PR? Can we accept it?

**Comment by rafamanzo** [comment]
After on a new look over I've found another bug introduced: hotspot metrics are getting listed under the tree ones list.

hotspot_list

Second, I would rename the `update_cache` private method to `clear_cache` for better semantics.

Third is good to notice that after fixing https://github.com/mezuro/kalibro_client/issues/85 the commit https://github.com/mezuro/prezento/commit/fe3e77175588035cdf9ed295ebb17d460bb5002f will no longer be necessary.

At last I've created a enhancement issue for this: https://github.com/mezuro/prezento/issues/288

**Comment by rafamanzo** [comment]
And I've one last question for all: does this PR make an upcoming release incompatible with the previous ones?

I'm tending to believe yes but I'm still gathering arguments :smile:

**Comment by dread-uo** [comment]
@diegoamc , I double-checked it, it is good to go :)
And I'm not sure about incompatibilities.

**Pull request merged**